### DISCUSSION TITLE
How to import vitepress settings ?

### DISCUSSION THREAD

**Discussion by Zogsha** [vuejs/vitepress#1392] [Q&A] [answered]
Hi,

I would like to import `outDir` to generate a sitemap. I know I can do it directly in `.vitepress/config.ts` and do something like `buildEnd : ({ outDir }) => ...` (#520) but I would like to generate my sitemap, pwa, etc in a `build.ts` file and just import my buildEnd script in `.vitepress/config.ts`, in order to declutter my config file.

I guess it's a trivial question, but I can't figure how to do it. Could you help me ?
Thanks

**Comment by brc-dd** [comment, marked as answer]
Something like this should work fine for you:

```ts
// some/script.ts
import { type SiteConfig } from 'vitepress'

export const buildEnd = ({ outDir } : SiteConfig) => {
  // ...
}
```

```ts
// .vitepress/config.ts
import { buildEnd } from 'path/to/some/script.js'

export default {
  // ...
  buildEnd 
}
```
